src/compiler/crystal/command.cr

```diff
@@ -130,6 +130,8 @@ class Crystal::Command
     else
       if command.ends_with?(".cr")
         error "file '#{command}' does not exist"
+      elsif external_command = Process.find_executable("crystal-#{command}")
+        Process.exec(external_command, options, env: {"CRYSTAL" => Process.executable_path})
```

Comment:
**issue:** `options` still includes the command as first entry. Without removing that, calling `crystal foo` would end up executing `crystal-foo foo`, wouldn't it?

Reply:
That's intentional. I copied that from cabal. I imagine it allows aliases and multiple commands with the same binary as long as it's symlinked with the additional alias/command.